xbmc/windowing/mir/WinEventsMir.cpp

```diff
@@ -329,10 +331,12 @@ bool CWinEventsMir::MessagePump()
 
 size_t CWinEventsMir::GetQueueSize()
 {
+  std::lock_guard<decltype(mutex)> event_lock(mutex);
```

Comment:
that's a nice c++11 feature which could replace most of our own locking patterns, that were implement this "by hand" when there was no std::mutex and no std::lock_guard available.